server/controllers/cohort.js

```diff
@@ -147,24 +149,24 @@ const getCohortDetails = (req, resp) => {
       }
 
       const {
-        ownerIds,
+        ownerIds: ownersCollection,
         _id,
         isArchived,
         description,
         name,
-        memberIds: membersData
+        memberIds: membersCollection
       } = doc;
 
       if (isArchived) {
         return resp.status(200).json({ _id, isArchived, name });
       }
 
-      const owners = ownerIds.map(owner => owner.id);
+      const ownerIds = ownersCollection.map(owner => owner.id);
```

Comment:
ownerIds could be an array with ids (no need populate ownerIds), we don't need owners objects and mapping them to ids

Reply:
yes, but I need as well the whole collection of users to respondWithCohortMembers with data like display_name for example, and I also need only ids.

So populating ownerIds array above it's necessary here.

Reply:
but you populate memberIds - where you have all users data (members and owners), there is no need to duplicate users data on ownerIds, ownerIds is needed to check users roles...